### PR TITLE
Added create-react-native-app support

### PR DESCRIPTION
**Pull request by shilman**
Issue: #1115 

## What I did

1. detect CRNA projects
2. simplify RN and CRNA templates by consolidating app into single
`index.js` instead of `index.ios.js` and `index.android.js`
3. notify user of extra setup steps in CLI
4. document CRNA idiosyncrasies in README

Many thanks to @orta @tmeasday for helping me learn RN/CRNA basics!
And @inyono for rocking the post that inspired this PR.

## How to test
```shell
yarn create react-native-app myapp
cd myapp
../path/to/storybook/lib/cli/bin/generate.js
# follow CLI / README instructions
```

### CODE REVIEW

File: app/react-native/readme.md [hunk 1]
```diff
@@ -21,22 +21,53 @@ npm -g i @storybook/cli
...
+
+This will get you up and running quickly, but then you lose your app!
+There are multiple options here. for example, you can export conditionally:
+
+```js
+import StorybookUI from './storybook';
+module.exports = __DEV__ ? StorybookUI : App;
+```
+
+Alternatively, `StorybookUI` is simply a RN `View` component that can be embedded anywhere in your RN application, e.g. on a tab or within an admin screen.
```
Comment (orta) [inline app/react-native/readme.md]: This is a good coverage of options, nice work

File: lib/cli/generators/REACT_NATIVE_SCRIPTS/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  const packageJson = helpers.getPackageJson();
+
+  packageJson.dependencies = packageJson.dependencies || {};
+  packageJson.devDependencies = packageJson.devDependencies || {};
+
+  packageJson.devDependencies['@storybook/react-native'] = `^${version}`;
+
+  if (!packageJson.dependencies['react-dom'] && !packageJson.devDependencies['react-dom']) {
+    packageJson.devDependencies['react-dom'] = '^15.5.4';
+  }
```
Comment (orta) [inline lib/cli/generators/REACT_NATIVE_SCRIPTS/index.js]: Nice one 👍

**Review by orta — APPROVED** [review]

**Comment by shilman** [comment]
@inyono Mind reviewing this PR? I actually came up with the `__DEV__` trick independently, but simply added it in the `README` docs. Let me know what you think. Happy to make changes!

**Comment by inyono** [comment]
Nice one! Will look over the PR at latest tomorrow :)

**Review by danielduan — APPROVED** [review]
I don't have a native environment to test but it looks good!

**Comment by shilman** [comment]
@danielduan if you're curious to test it out, it takes just 3 minutes using the test instructions above -- super easy setup!

**Review by ndelangen — APPROVED** [review]

**Pull request merged**